pkg/controller/controller.go

```diff
@@ -180,6 +198,15 @@ func AddToManager(m manager.Manager, deps *Dependencies) error {
 		if a2, ok := a.(PubsubInjector); ok {
 			a2.InjectPubsubSystem(deps.PubsubSystem)
 		}
+		if a2, ok := a.(EventsChInjector); ok {
+			// this is used by the config controller to sync
+			a2.InjectEventsCh(events)
```

Comment:
I'm feeling ambivalent about exposing this events channel as one of the global dependencies.  This leads to the registrar that should be internal to the CacheManager being created in the config_controller.  In a future world where the config_controller and the syncset_controller are equal consumers of the CacheManager, why should one be the instantiator of the CM?  It feels odd.

While it was a little clunky that the sync_controller wasn't instantiated the same way as the other controllers (aka it wasn't included in the `Injectors` slice that's run through in `controller.go`), there was actually a benefit to this: the sync_controller was hidden from other parts of the system.  It was abstracted away.

I believe that we may actually be in a new world where it's an even _better_ idea to hide the `sync_controller` beneath a layer of abstraction.  

No other component is meant to interact with the sync_controller besides the CacheManager.  Having the CM control the instantiation of the sync_controller seems a fitting way to express that.

That said, both the CacheManager and the sync_controller _do_ use some of the dependencies are injected into the controllers.  Is that the only way to obtain those dependencies?